### PR TITLE
Variant composition

### PR DESCRIPTION
**Pull request by edawson**
Finishes initial naive implementation of variant composition for indels.
- [ ] Needs tests.
- [ ] Needs more speed (probably primarily by minimizing allocs under the hood).

### CODE REVIEW

File: src/nibpkg/compose.nim [hunk 1]
```diff
@@ -6,15 +6,35 @@ type
...
-  ## function to compose
+  return combined_sequence
+
+
+proc compose_variants*(variant_file: string, reference_file: string): seq[string] =
+  ## function to compose variants from their sequence / FASTA flanking regions
+  ## Returns a Sequence of strings representing the DNA sequence of the flanking
+  ## regions and variant sequence.
+
+  var composed_seqs = newSeq[string](1)
```
Comment (pb-cdunn) [inline src/nibpkg/compose.nim]: `newSeq[string]()`

It can start out empty. The `.add()` function extends the array by 1.

**Review by pb-cdunn — APPROVED** [review]
Aside from that, looks good to me.

File: src/nibpkg/compose.nim [hunk 2]
```diff
@@ -6,15 +6,35 @@ type
...
+  var inner_seq: string
+  var variant_type: string
+  doAssert variant.info.get("SVTYPE", variant_type) == Status.OK
+  if variant_type == "DEL":
+    inner_seq = ""
+  elif variant_type == "INS":
+    inner_seq = variant.ALT[0]
+  elif variant_type == "INV":
+    raise newException(ValueError,
+    "Error: Inversion processing not implemented.")
```
Comment (zeeev) [inline src/nibpkg/compose.nim:26]: nice.

Comment (edawson) [reply src/nibpkg/compose.nim:26]: Every dang time - we'll implement those later!

**Pull request merged**